doc/faq.rst

```diff
@@ -264,12 +264,7 @@ collection, configured to use :class:`~bson.son.SON` instead of dict:
   >>> from bson import CodecOptions, SON
   >>> opts = CodecOptions(document_class=SON)
   >>> opts
-  CodecOptions(document_class=<class 'bson.son.SON'>,
-               tz_aware=False,
-               uuid_representation=UuidRepresentation.UNSPECIFIED,
-               unicode_decode_error_handler='strict',
-               tzinfo=None, type_registry=TypeRegistry(type_codecs=[],
-                                                       fallback_encoder=None))
+  CodecOptions(document_class=bson.son.SON, tz_aware=False, uuid_representation=UuidRepresentation.UNSPECIFIED, unicode_decode_error_handler='strict', tzinfo=None, type_registry=TypeRegistry(type_codecs=[], fallback_encoder=None))
```

Comment:
Is this a bug or does it have something to do with inheriting from SON now inheriting `typing.Dict`? On Python 3.6:
```
>>> repr(SON)
'bson.son.SON'
```
Vs Python 3.7+
```
>>> repr(SON)
"<class 'bson.son.SON'>"
```

Should we bother trying to make this consistent?

Reply:
In Python 3.7+, the built in container types are generic as well.   I don't think it is worth supporting the Python 3.6 syntax in the doctests.

Reply:
Oh I see:
```
>>> repr(SON)
"<class 'bson.son.SON'>"
>>> repr(SON[str, int])
'bson.son.SON[str, int]'
>>> repr(MongoClient)
"<class 'pymongo.mongo_client.MongoClient'>"
>>> repr(MongoClient[dict])
'pymongo.mongo_client.MongoClient[dict]'
```

Sees odd that they didn't stick with one representation. There's probably a related CPython or Mypy bugs report but I can't find it at the moment.